src/exec/DB.hs

```diff
@@ -124,6 +131,9 @@ withDB dbOptions dbTracer indexTracer rr nodeConfig extLedgerState k = do
         , cdbGenesis = pure extLedgerState
 
         , cdbTracer = dbTracer
+        , cdbTraceLedger = case traceLedgerEvery dbOptions of
+                             Nothing -> Tracer.nullTracer
+                             Just n  -> traceLedgerDbSize (\slot -> slot `mod` n == 0) traceDbSize
```

Comment:
This is an interesting use case. We've ostensibly got 2 degrees of freedom:

1. The number `n :: Word64` of slots to wait between _actually_ tracing the size.
2. The tracer itself `traceDbSize :: Tracer IO (LedgerDbSize (Block ByronConfig))`.

IMO they're actually the same thing, as far as the _author_ of this definition `withDB` should be concerned. If the _user_ of `withDB` wishes to trace every `n` slots, or not trace at all, that question is in the same domain as whether they want to trace to stdout versus to a file: it should be expressed entirely by the `Tracer`.

How would that look... the caller would give a `traceDbSize :: Tracer IO (LedgerDbSize (Block ByronConfig))`, and we'd hope that this could be `contramap`ped or similar into a `Tracer IO (LedgerDB l (Point ByronConfig))` such that the `LedgerDbSize` is not forced unless `traceDbSize` forces it. But that's not so simple, because computing the size needs `IO`. If it were (is it?) safe to compute the sizes in `unsafeInterleaveIO` then this world work. Or if the `LedgerDbSize` record contained `IO (Either CountFailure Word64)`, that would work too.

So the reason that the number `n` and the tracer `traceDbSize` are kept apart is just because we need to check the tip slot before computing the `LedgerDbSize`.

Another thing worth mentioning is that what we really want (maybe I'm wrong?) is to trace every `n` _changes_ to the tip of chain, whereas this implementation only traces on certain slots, and indeed may never trace at all, since not all slots are necessarily filled. This could of course be solved with a kind of "count down" higher-order tracer:

```Haskell
-- Only traces on the n'th trace. Could be used in multiple threads but that would be odd.
countdown :: Word64 -> Tracer IO a -> IO (Tracer IO a)
countdown n tr = do
  cref <- newIORef n
  pure $ Tracer $ \a -> do
    shouldTrace <- atomicModifyIORef' $ \m -> if m == 0 then (n, True) else (n-1, False)
    when shouldTrace (traceWith tr a)
```

Anyway, just thinking aloud...